### PR TITLE
VACMS-5650: Knowledge Base iteration.

### LINKED ISSUE #5718: Help Center search Proof of Concept

### LINKED ISSUE #5720: Implement Help Center search

### PR DESCRIPTION
**Pull request by kevwalsh**
## Description

Closes #5720 #5804 #5806 #5729 #5814 #5721 #5849 #5913. 

This is a feature branch with multiple updates to Knowledge Base (including its rebranding from Help Center).

Requires some content updates post deploy, each time code changes.  

## Testing done

Behat and Phpunit (tests coming soon via https://github.com/department-of-veterans-affairs/va.gov-cms/issues/5919)

Visual

## Post deploy steps

- [x] [Re-index Knowledge Base](https://pr5914-i1b3k13mkin6pssflrlbeopjpxlgmwei.ci.cms.va.gov/admin/config/search/search-api/index/knowledge_base_search)  
- [x] [Archive current homepage](https://pr5914-i1b3k13mkin6pssflrlbeopjpxlgmwei.ci.cms.va.gov/node/6454/edit) 
  - [ ] Change URL to /help-old
  - [ ] Save as archived
- [x] Edit [node 26858](https://pr5914-i1b3k13mkin6pssflrlbeopjpxlgmwei.ci.cms.va.gov/node/26856/edit) to make it the new KB home
  - [ ] Change URL to /help 
  - [ ] Move menu item to root of KB menu
  - [ ] Save as Published
- [x] [Move children of old homepage](https://pr5914-i1b3k13mkin6pssflrlbeopjpxlgmwei.ci.cms.va.gov/admin/structure/menu/manage/documentation) to children of new homepage 
- [x] Add search block at the top of KB homepage
  - [ ]  Go to [node/26856/layout](https://pr5914-i1b3k13mkin6pssflrlbeopjpxlgmwei.ci.cms.va.gov/node/26856/layout )
  - [ ] add the block "Exposed form: knowledge_base_search_results-knowledge_base_search_page" and position it at the top of the page, and save. 
- [x] Bulk update KB articles
  - [ ] Go to [new KB admin page](https://pr5914-i1b3k13mkin6pssflrlbeopjpxlgmwei.ci.cms.va.gov/admin/content/audit/knowledge-base-article-administration)
  - [x] Update Product field for a few KB articles with "Modify field values"
  - [ ] Update URL aliases to pick up the new IA. Check all nodes except the /help homepage and the /help/vamcs (we want to keep those as they are, with their custom URL aliases), then "Update URL alias". 
  - [ ] Publish latest revision of all non-archived content. 
 
## QA steps

**Architectural review**
- [x] Search api index has no performance concerns
- [x] Layout builder config 

**Functional QA**
- [ ] Perform post deploy steps above
- [x] #5720 
  - [ ] Go to /help 
  - [ ] Search for various terms and review results, such as "video", "health", or "link". these terms should appear with the `keyword` parameter (like `/help/search?keyword=video`)
  - [ ] Validate that filters work as expected, and have a reset button on the search results page
  - [ ] Validate that you can get back to the KB homepage via the breadcrumb
  - [ ] Validate that search filters are available on all pages on the left nav, with the exception of /help, where it's at the top.
- [ ] #5804 
  - [ ] Validate that the ? icon in the toolbar menu now has a "Knowledge Base" label beside it, not "Help Center"
- [ ] #5806 
  - [ ] Go to a variety of KB articles and validate that "Related user guides" now just says "Related" at the bottom. 
- [ ] #5729 
  - [ ] Validate that the [enabled KB menu items](https://pr5914-i1b3k13mkin6pssflrlbeopjpxlgmwei.ci.cms.va.gov/admin/structure/menu/manage/documentation) match the menu items in the left nav
  - [ ] Validate that the left nav only is shown on KB articles other than /help
  - [ ] Validate that only the enabled top level menu items are shown
- [ ] #5814 
  - [ ] Validate that the [KB admin page](https://pr5914-i1b3k13mkin6pssflrlbeopjpxlgmwei.ci.cms.va.gov/admin/content/audit/knowledge-base-article-administration) shows as much relevant content that fits, including the URL alias for each KB article
- [ ] #5721 
  - [ ] Verify that [KB articles can now be tagged by product](https://pr5914-i1b3k13mkin6pssflrlbeopjpxlgmwei.ci.cms.va.gov/node/26856/edit) and that it's optional
  - [ ] Verify that product metadata appears on the [KB admin page](https://pr5914-i1b3k13mkin6pssflrlbeopjpxlgmwei.ci.cms.va.gov/admin/content/audit/knowledge-base-article-administration)
- [ ] #5849 
  - [ ] Validated that KB articles how show Last updated at the bottom. 
- [ ] #5913
   - [ ] Validate (via post-deploy steps) that any KB article can have its layout overridden via the Layout tab

**FE QA**
- [ ] As a non-admin user, QA new UI elements (search, navigation) across supported browsers (Edge, Chrome, FF)

**Security / permission review**
- [ ] Log in as a non-content admin or admin and attempt all QA steps below. Any KB admin or editing QA steps should fail. 

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [x] `Core Application Team`
- [x] `Product Support Team`

### Is this PR blocked by another PR?

- [x] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [x] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change? 
- [x] Yes, and it's written in https://docs.google.com/presentation/d/1wxWDGRIf-RbZwQXB0krdMUGfli4cQHRsCWLguhCUR6Y/edit#slide=id.gd15cb54f12_1_0 
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written 
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change. 
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

**Comment by ndouglas** [comment]
The test definition that's causing the failure is in `tests/behavioral/cypress/integration/common/i_create_a_node.js`.

On line 11: 
```
    cy.findAllByLabelText('Parent link').select('-- CMS help center', { force: true });
```

should presumably be something like 

```
    cy.findAllByLabelText('Parent link').select('-- CMS Knowledge Base', { force: true });
```

But currently, I think that says "-- CMS Knowledge Base (disabled)" because it's a work in progress.  If that's going to change a time or two, I think it's safe to just remove the line completely and we can add it back in once it's settled.  I suspect (though I haven't checked) that the line is just there for aesthetics and consistency and not really a functional requirement of the test.

**Comment by ndouglas** [comment]
Thinking past that a little bit, if the What's New in the CMS block is going away, there's quite a bit of cruft that can be stripped out safely.  The last three tests from `tests/behavioral/cypress/integration/help_center.feature` and the `tests/behavioral/cypress/integration/help_center/whats_new_in_the_cms_block.js` file as well.  I'll just open a PR based off this PR tomorrow doing that, as you suggested 😃

File: .phpunit.result.cache [hunk 1]
```diff
@@ -0,0 +1 @@
...t nodes in their sections\"":1.94,"tests\\phpunit\\Security\\SectionAccessTest::testSectionAccess with data set \"Content editors may not edit nodes in other sections\"":2.077,"tests\\phpunit\\Security\\WorkflowTest::testAccessWorkflow":1.769,"tests\\phpunit\\Service\\EditorialWorkflowContentRepositoryServiceTest::testGetLatestArchivedRevisionId":2.201,"tests\\phpunit\\Service\\EditorialWorkflowContentRepositoryServiceTest::testGetLatestPublishedRevisionId":2.689,"tests\\phpunit\\Service\\ExclusionTypesServiceTest::testGetExcludedTypes":0.012,"tests\\phpunit\\Service\\ExclusionTypesServiceTest::testGetJson":0,"tests\\phpunit\\Service\\ExclusionTypesServiceTest::testTypeIsExcluded":0.001,"tests\\phpunit\\Service\\ModerationActionsServiceTest::testArchiveNode":1.686,"tests\\phpunit\\Service\\ModerationActionsServiceTest::testPublishLatestRevision":1.871,"tests\\phpunit\\Service\\VaGovUrlServiceTest::testGetVaGovFrontEndUrl":0.7,"tests\\phpunit\\Service\\VaGovUrlServiceTest::testOverrideVaGovFrontEndUrl":0.69,"tests\\phpunit\\Service\\VaGovUrlServiceTest::testGetVaGovFrontEndUrlForEntity":2.899,"tests\\phpunit\\Service\\VaGovUrlServiceTest::testVaGovFrontEndUrlForEntityIsLive":1.808}}
```
Comment (swirtSJW) [inline .phpunit.result.cache:1]: Is this intentional?

**Review by swirtSJW — COMMENTED** [review]
All looks good.  Just the one question

File: .phpunit.result.cache (hunk 1 above)
Comment (ethanteague) [reply .phpunit.result.cache:1]: Not intentional, @swirtSJW

**Review by swirtSJW — APPROVED** [review]

File: .phpunit.result.cache (hunk 1 above)
Comment (ndouglas) [reply .phpunit.result.cache]: https://github.com/department-of-veterans-affairs/va.gov-cms/pull/5938 just to prevent this longterm FYI

**Pull request merged**